### PR TITLE
Added documentation for Coverage-By-Encounter Write 

### PR DESCRIPTION
**Pull request by ahamdany**
Changes:
* Added encounter request body.
* Added encounter-extension body field.

Local Verification:

Screen Shot 2020-03-24 at 4 35 11 PM

Screen Shot 2020-03-24 at 4 36 45 PM

### CODE REVIEW

File: lib/resources/example_json/r4_examples_coverage.rb [hunk 1]
```diff
@@ -281,5 +282,76 @@ module Resources
...
+      "extension": [
+      {
+        "url": "https://fhir-ehr.cerner.com/r4/StructureDefinition/coverage-encounter",
+        "valueReference": {
+          "reference": "Encounter/31363178"
+      }
+    }
+    ],
+      "subscriberId": "1626016",
+      "dependent": "1626016",
```
Comment (MadhurRaj) [inline lib/resources/example_json/r4_examples_coverage.rb]: can we update these, if they don't have to be the same as the beneficiary id?

File: content/millennium/r4/financial/coverage.md [hunk 2]
```diff
@@ -119,10 +119,14 @@ Create a new Patient-level Coverage.
 
     POST https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Coverage
 
-#### Body
+#### Patient-Body
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply content/millennium/r4/financial/coverage.md]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/8e93680b7b622bf5b2dec1f2f344e46d073b2a75#diff-81b4004b2ae7fd9a288412597630810cL100)

File: content/millennium/r4/financial/coverage.md [hunk 3]
```diff
@@ -119,10 +119,14 @@ Create a new Patient-level Coverage.
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply content/millennium/r4/financial/coverage.md:119]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/8e93680b7b622bf5b2dec1f2f344e46d073b2a75#diff-81b4004b2ae7fd9a288412597630810cL100)

File: lib/definition_table_generator.rb [hunk 4]
```diff
@@ -13,6 +13,10 @@ def patch_definition_table(content_key, version)
   DefinitionTableGenerator.new(version, :patch).patch_definition_table(content_key)
 end
 
+# def write_definition_table(content_key, version)
+#   DefinitionTableGenerator.new(version, :patch).patch_definition_table(content_key)
+# end
+
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply lib/definition_table_generator.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/8e93680b7b622bf5b2dec1f2f344e46d073b2a75#diff-73e703ab5c2efa2a629ba1622a8cd56dL16)

File: lib/resources/example_json/r4_examples_coverage.rb [hunk 5]
```diff
@@ -227,6 +227,7 @@ module Resources
       ]
     }
 
+
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_coverage.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/8e93680b7b622bf5b2dec1f2f344e46d073b2a75#diff-398829d7a4d0e757ca54b5f56e50d99aL230)

File: lib/resources/example_json/r4_examples_coverage.rb [hunk 6]
```diff
@@ -281,5 +282,76 @@ module Resources
         }
       }
     ]
+
+    R4_COVERAGE_ENCOUNTER_CREATE ||= {
+      "resourceType": "Coverage",
+      "status": "active",
+      "subscriber": {
+        "reference": "Patient/1626016",
+      "display": "WOLFE, STEPHANIE T"
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_coverage.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/8e93680b7b622bf5b2dec1f2f344e46d073b2a75#diff-398829d7a4d0e757ca54b5f56e50d99aL286)

File: lib/resources/r4/coverage.yaml [hunk 7]
```diff
@@ -15,6 +15,23 @@ fields:
   note: The resourceType must be Coverage.
   url: http://hl7.org/fhir/r4/json.html#resources
 
+- name: Coverage Encounter Extension
+  required: 'No'
+  cardinality: 0..1
+  type: Extension
+  description: Reference to the Encounter associated to the encounter level Coverage.
```
Comment (MadhurRaj) [inline lib/resources/r4/coverage.yaml:22]: If we're using Encounter-level Coverage, we should use it consistently.

Comment (ahamdany) [reply lib/resources/r4/coverage.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/cf1c358e4663e9996a56a620adadf3a72acab2c7#diff-af9c966415db83d073f1bcec5ed6f875L22)

File: lib/resources/r4/coverage.yaml [hunk 8]
```diff
@@ -15,6 +15,23 @@ fields:
...
+  note:
+    - The <code>valueReference.reference</code> must be an Encounter reference.
+    - If this extension is present an encounter level coverage will be posted.
```
[the remark this answers is not in the document]
Comment (ahamdany) [reply lib/resources/r4/coverage.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/cf1c358e4663e9996a56a620adadf3a72acab2c7#diff-af9c966415db83d073f1bcec5ed6f875L25)

File: lib/resources/example_json/r4_examples_coverage.rb (hunk 1 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_coverage.rb]: These ids have been used in all postman calls, so I'm not sure if they are suppose to be different.

File: content/millennium/r4/financial/coverage.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/millennium/r4/financial/coverage.md]: change this to "Example - Encounter-level Coverage" and change the other to "Example - Patient-level Coverage"

break out h4's for Request and Response inside both sections

File: lib/resources/example_json/r4_examples_coverage.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply lib/resources/example_json/r4_examples_coverage.rb]: 64de437

File: content/millennium/r4/financial/coverage.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahamdany) [reply content/millennium/r4/financial/coverage.md]: updated again [here](https://github.com/cerner/fhir.cerner.com/commit/c4a5b1f1c13a36bdb70e6e722e937e3b4f379daa#diff-81b4004b2ae7fd9a288412597630810cL116)

File: lib/resources/example_json/r4_examples_coverage.rb (hunk 1 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_coverage.rb]: updated [here](https://github.com/cerner/fhir.cerner.com/commit/f0575498e7ea66b3c22a8540b3ef055db4af122d#diff-398829d7a4d0e757ca54b5f56e50d99aL300)

File: content/millennium/r4/financial/coverage.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/millennium/r4/financial/coverage.md]: f63de2d

File: content/millennium/r4/financial/coverage.md [hunk 9]
```diff
@@ -147,6 +147,40 @@ X-Runtime: 3.890282
...
+<pre class="terminal">
+Cache-Control: no-cache
+Content-Length: 0
+Content-Type: text/html
+Date: Tue, 22 Oct 2019 15:59:33 GMT
+Etag: W/"0"
+Last-Modified: Tue, 22 Oct 2019 15:59:30 GMT
+Location: https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Coverage/31363178-11500257
+Server-Response-Time: 3890.363996
+Status: 201 Created
```
Comment (MadhurRaj) [inline content/millennium/r4/financial/coverage.md:172]: Let's remove the status header.

File: content/millennium/r4/financial/coverage.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply content/millennium/r4/financial/coverage.md]: TOC rendering fixes https://github.com/cerner/fhir.cerner.com/pull/357/commits/ec65adaf23ee1ac3262d7653afda097101a198f4#diff-81b4004b2ae7fd9a288412597630810cR122

File: content/millennium/r4/financial/coverage.md (hunk 9 above)
Comment (MaxPhilips) [reply content/millennium/r4/financial/coverage.md]: https://github.com/cerner/fhir.cerner.com/pull/357/commits/ec65adaf23ee1ac3262d7653afda097101a198f4#diff-81b4004b2ae7fd9a288412597630810cL138

File: lib/resources/r4/coverage.yaml [hunk 10]
```diff
@@ -15,6 +15,23 @@ fields:
...
   url: http://hl7.org/fhir/r4/json.html#resources
 
+- name: Coverage Encounter Extension
+  required: 'No'
+  cardinality: 0..1
+  type: Extension
+  description: Reference to the Encounter associated to the Encounter-level Coverage.
+  note:
+    - The <code>valueReference.reference</code> must be an Encounter reference.
+    - If provided, an Encounter-level Coverage will be created.
```
Comment (MaxPhilips) [inline lib/resources/r4/coverage.yaml:25]: note formatting must be achieved via inline html, not yaml

Comment (MaxPhilips) [reply lib/resources/r4/coverage.yaml]: 2b77ece

**Comment by MaxPhilips** [comment]
validated recent changes

Screen Shot 2020-04-07 at 4 38 29 PM

Screen Shot 2020-04-07 at 4 38 45 PM

**Comment by ebaranowski** [comment]
Screen Shot 2020-04-07 at 4 39 37 PM
Screen Shot 2020-04-07 at 4 39 45 PM
Screen Shot 2020-04-07 at 4 39 52 PM

**Comment by ebaranowski** [comment]
Validated: 

Screen Shot 2020-04-07 at 5 21 31 PM
Screen Shot 2020-04-07 at 5 21 53 PM

**Pull request merged**